pkg/inframetadata/gohai/gohai.go

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
I don't feel like this qualifies as a separate package. Actually, I think all you need is one package for all of this. None of these packages need to be re-used elsewhere, nor are they individually useful. A package is not meant for separating logic, it's supposed to have a self-contained scope and be usable by at least two other packages.